src/ray/design_docs/actor_states.rst

```diff
@@ -0,0 +1,48 @@
+Actor State: Definitions & Transition Diagram
+============================================
+
+An actor can be in one of the following states:
+
+- **DEPENDENCIES_UNREADY**: the actor info ins registered in GCS. But its dependencies are not ready.
+
+- **PENDING_CREATION**: the actor local dependencies are ready. This actor is being created.
+
+- **ALIVE**: the actor is created successfully.
+
+- **RESTARTING**: the actor is dead, now being restarted. After reconstruction finishes,
+  the state will become alive again.
+
+- **DEAD**: the actor is already dead and won't be restarted.
+
+::
+
+                                                                         3
+   0                            1                       2          ------------->
+ ---->DEPENDENCIES_UNREADY-------->PENDING_CREATION-------->ALIVE                RESTARTING
+               |                            |                  |   <-------------      |
+               |                            |                  |         4             |
+               |                            |                  |                       |
+             8 |                         7  |                6 |                       | 5
+               |                            |                  |                       |
+               |                            |                  |                       |
+               |                            |                  |                       |
+               |                            v                  |                       |
+                -------------------------->DEAD<---------------------------------------
+
+- **0**: GCS will create an actor whose state is `DEPENDENCIES_UNREADY` when a `RegisterActor` request is received from CoreWorker.
```

Comment:
```suggestion
- **0**:  When GCS receives a `RegisterActor` request from core worker, GCS will persist the actor in database with state `DEPENDENCIES_UNREADY`.
```

Reply:
Nit: 
1) I think it's clearer to state what event will trigger the state transition first.
2) No need to use camel case for a term (e.g., `CoreWorker`), unless that is something you can search in code (e.g, `RegisterActor`).

Reply:
fixed